### DISCUSSION TITLE
实时编辑primitive出现报错

### DISCUSSION THREAD

**Discussion by MangMax** [zouyaoji/vue-cesium#383] [Q&A] [answered]
我想编辑primitive并更新，使用getCesiumObject()获取vc-primitive组件的cesium对象，并调用update()方法，出现了报错
![image](https://user-images.githubusercontent.com/68525017/235642424-b9e88f0e-7098-4bb8-a218-44c7fe8d1fa5.png)
![image](https://user-images.githubusercontent.com/68525017/235642643-9c8a1ee0-2c0d-4fe0-918b-bb669640bd72.png)
是不能用这种方式更新图元吗，同时我还发现获取到的对象geometryInstances是空的，但是地图上其实是加载了图元的，确认是有值的
![image](https://user-images.githubusercontent.com/68525017/235642970-947adc8d-ed3b-4794-9891-b8d59ece3f96.png)
我看文档中vc-primitive没有支持任何方法，望解答

**Comment by zouyaoji** [comment, marked as answer]
primitive没法直接改的吧，都是先移除再添加。

geometryInstances是空的是cesium默认优化掉了，具体你看看`releaseGeometryInstances` 参数
